src/refiner/development.fun

```diff
@@ -100,13 +100,13 @@ struct
          fun eq (l, r) = compare (l, r) = EQUAL
        end)
 
-  type object = Object.t
-  type world = {context : object Telescope.telescope,
-                resources : tactic list ResourcePool.dict}
+  datatype world = World of {context : world Object.t Telescope.telescope,
+                             resources : tactic list ResourcePool.dict}
+  type object = world Object.t
```

Comment:
I think it may be possible to have `context : object Telescope.telescope` if you change `type object = ...` to `withtype object = ...`.